libraries/aws/shadow/src/shadow.c

```diff
@@ -0,0 +1,618 @@
+/**
+ * @brief The string representing "/shadow/delete/accepted".
+ */
+#define SHADOW_OP_DELETE_REJECTED               SHADOW_OP_DELETE SHADOW_SUFFIX_REJECTED
+
+/**
+ * @brief The string representing "/shadow/get/accepted".
+ */
+#define SHADOW_OP_GET_ACCEPTED                  SHADOW_OP_GET SHADOW_SUFFIX_ACCEPTED
+
+/**
+ * @brief The string representing "/shadow/get/accepted".
+ */
+#define SHADOW_OP_GET_REJECTED                  SHADOW_OP_GET SHADOW_SUFFIX_REJECTED
+
+/**
+ * @brief The length of "/shadow/update/accepted".
+ */
+#define SHADOW_OP_UPDATE_ACCEPTED_LENGTH               ( SHADOW_OP_UPDATE_LENGTH + SHADOW_SUFFIX_ACCEPTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/update/rejected".
+ */
+#define SHADOW_OP_UPDATE_REJECTED_LENGTH               ( SHADOW_OP_UPDATE_LENGTH + SHADOW_SUFFIX_REJECTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/update/document".
+ */
+#define SHADOW_OP_UPDATE_DOCUMENTS_LENGTH              ( SHADOW_OP_UPDATE_LENGTH + SHADOW_SUFFIX_DOCUMENTS_LENGTH )
+
+/**
+ * @brief The length of "/shadow/update/rejected".
+ */
+#define SHADOW_OP_UPDATE_DELTA_LENGTH                  ( SHADOW_OP_UPDATE_LENGTH + SHADOW_SUFFIX_DELTA_LENGTH )
+
+/**
+ * @brief The length of "/shadow/get/accepted".
+ */
+#define SHADOW_OP_GET_ACCEPTED_LENGTH                  ( SHADOW_OP_GET_LENGTH + SHADOW_SUFFIX_ACCEPTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/get/rejected".
+ */
+#define SHADOW_OP_GET_REJECTED_LENGTH                  ( SHADOW_OP_GET_LENGTH + SHADOW_SUFFIX_REJECTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/get/accepted".
+ */
+#define SHADOW_OP_DELETE_ACCEPTED_LENGTH               ( SHADOW_OP_DELETE_LENGTH + SHADOW_SUFFIX_ACCEPTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/delete/rejected".
+ */
+#define SHADOW_OP_DELETE_REJECTED_LENGTH               ( SHADOW_OP_DELETE_LENGTH + SHADOW_SUFFIX_REJECTED_LENGTH )
+
+/**
+ * @brief Determine if the string contains the substring.
+ *
+ * @param[in] pString Pointer to the string.
+ * @param[in] stringLength Length of pString.
+ * @param[in] pSubString Pointer to the substring.
+ * @param[in] subStringLength Length of pSubString.
+ *
+ * @return Return SHADOW_SUCCESS if it contains;
+ *         return SHADOW_FAIL if not.
+ */
+static ShadowStatus_t containsSubString( const char * pString,
+                                         uint16_t stringLength,
+                                         const char * pSubString,
+                                         uint16_t subStringLength );
+/**
+ * @brief Extract the Thing Name from a string.
+ *
+ * @param[in] pString Pointer to the starting of thing name.
+ * @param[in] stringLength Length of pString.
+ * @param[out] pThingNameLength Pointer to caller-supplied memory for returning the length of the Thing Name.
+ *
+ * @return Return SHADOW_SUCCESS if successfully extracted;
+ *         return SHADOW_THINGNAME_PARSE_FAILED if failed.
+ */
+static ShadowStatus_t extractThingName( const char * pString,
```

Comment:
Given the changes we made to this function, perhaps it should be renamed to something like `determineThingNameLength` since it extracts the length instead of the string itself

Reply:
make sense, I'll change it.
Or how about validate thing Name? Since it is checking if the thing name is valid, exists or not.

Let me know which one you prefer

Thanks

Reply:
I don't have a preference, either one is fine with me

Reply:
fixed it in commit [9f0e875](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1081/commits/9f0e8750998f85c05b8bbb1ad475ea7cee3ad124)

by renaming extractThingName to validateThingName and changed the comments.

resolving this comment

feel free to let me know if it needs further changes, thank you